neuralop/layers/neighbor_search.py

```diff
@@ -61,31 +69,34 @@ def forward(self, data, queries, radius):
             return_dict['neighbors_row_splits'] = search_return.neighbors_row_splits.long()
 
         else:
-            neighbors_count, neighbors_index = self.search_fn(data, queries, radius,
-                                               max_num_neighbors=data.shape[0])
-            
-            if neighbors_count[-1] != queries.shape[0] - 1:
-                add_max_element = True
-                neighbors_count = torch.cat((neighbors_count,
-                                             torch.tensor([queries.shape[0] - 1], 
-                                                          dtype=neighbors_count.dtype,
-                                                          device=neighbors_count.device)), 
-                                                          dim=0)
+            if not self.use_torch_cluster:
+                return_dict = self.search_fn(data, queries, radius)
```

Comment:
@kovachki Should we just keep the manual implementation and just choose between open3d (fast) and fallback (slow)?

Reply:
Implemented.

Reply:
Yes, I think this is best. Let's just remove cluster altogether.